opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1016,12 +1014,15 @@ namespace Opm {
                 const unsigned cellIdx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& intQuants = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& fs = intQuants.fluidState();
+                const double pv =
+                    ebosSimulator_.model().dofTotalVolume(cellIdx)
+                    * ebosSimulator_.problem().porosity(cellIdx);
```

Comment:
While I am perhaps clinging a bit close to the Eclipse-like way of thinking here, I think of (or have over time become conditioned to it) pore volume as consisting of a reference pore volume and a dynamic multiplier (pressure dependence). And in computing the reference pore volume there are lots of nitty-gritty details, that are not reflected in the apparent formula used here (volume times porosity). If it is correct for this particular purpose, then it must be because either the volume or the porosity must include such things as NTG. I assume it is baked into the porosity. In that case it will be wrong for various polymer simulations for example, which include an adsorption term that depends on porosity.

To sum up, I really don't like the explicit use of a formula here to compute pore volume, where before we used an already-digested number.